### ISSUE TITLE
Evaluate and Select Appropriate Loss Function and Optimizer

### ISSUE THREAD

**Issue by yashwantherukulla** [yashwantherukulla/SnakeSpotter#4]
The current choice of loss function and optimizer may not be optimal this particular task. Investigate alternative loss functions and optimizers that could potentially improve model training and performance and adjust the related hyperparameters however necessary.

Prerequisites: #3
